pkg/pillar/dpcreconciler/linuxitems/route.go

```diff
@@ -44,19 +52,61 @@ func (r Route) Label() string {
 	} else {
 		dst = r.Route.Dst.String()
 	}
-	return fmt.Sprintf("IP route table %d dst %s dev %v via %v",
-		r.Table, dst, r.AdapterLL, r.Gw)
+	return fmt.Sprintf("%s route table %d dst %s dev %v via %v",
+		r.ipVersionStr(), r.Table, dst, r.AdapterLL, r.Gw)
+}
+
+func (r Route) ipVersionStr() string {
+	switch r.Family {
+	case netlink.FAMILY_V4:
+		return "IPv4"
+	case netlink.FAMILY_V6:
+		return "IPv6"
+	default:
+		return "IPv?"
+	}
+}
+
+func (r Route) hasDefaultDst() bool {
+	if r.Route.Dst == nil {
+		return true
+	}
+	ones, _ := r.Route.Dst.Mask.Size()
+	return ones == 0 && r.Route.Dst.IP.IsUnspecified()
+}
+
+// Any destination IP and nil destination IP are treated as the same.
+// However, netlink RouteAdd and RouteDel require a non-nil destination IP.
+func (r Route) normalizedNetlinkRoute() netlink.Route {
+	route := r.Route
+	if route.Dst == nil {
+		if route.Family == netlink.FAMILY_V4 {
+			route.Dst = ipv4Any
+		} else {
+			route.Dst = ipv6Any
+		}
+	}
+	// Also clear flags like RTNH_F_LINKDOWN - in the scope of *config* reconciliation
+	// we do not care about them.
+	route.Flags = 0
+	return route
 }
 
 // Type of the item.
 func (r Route) Type() string {
-	return genericitems.RouteTypename
+	if r.Family == netlink.FAMILY_V6 {
+		return genericitems.IPv6RouteTypename
+	}
+	return genericitems.IPv4RouteTypename
 }
 
 // Equal is a comparison method for two equally-named route instances.
 func (r Route) Equal(other depgraph.Item) bool {
-	r2 := other.(Route)
-	return r.Route.Equal(r2.Route)
+	r2, isRoute := other.(Route)
+	if !isRoute {
+		return false
```

Comment:
Why do you need this? To handle a nil other? Comment might be helpful.

Reply:
This is actually not needed as it is unreachable. I only added this to get rid of warning of some of the Yetus checks ("nil check missing" or something like that).
And in the upcoming patch there is a lot more of this. So either we decide to ignore these failed checks or we put these unreachable branches. But I would prefer not to spam the code with "for yetus" comments.

Reply:
OK